guacamole-ext/src/main/java/org/apache/guacamole/net/event/AuthenticationSuccessEvent.java

```diff
@@ -43,19 +44,19 @@
     /**
      * The credentials which passed authentication.
      */
-    private Credentials credentials;
+    private AuthenticatedUser authenticatedUser;
 
     /**
      * Creates a new AuthenticationSuccessEvent which represents a successful
      * authentication attempt with the given credentials.
      *
      * @param context The UserContext created as a result of successful
      *                authentication.
-     * @param credentials The credentials which passed authentication.
+     * @param authenticatedUser The user which passed authentication.
      */
-    public AuthenticationSuccessEvent(UserContext context, Credentials credentials) {
+    public AuthenticationSuccessEvent(UserContext context, AuthenticatedUser authenticatedUser) {
```

Comment:
You could drop `UserContext` in the constructor here and drop the associated field. Then you could implement `getUserContext` as:

```
public UserContext getUserContext() {
  return authenticatedUser.getAuthenticationProvider().getUserContext(authenticatedUser);
}
```

This will avoid the need to handle the exception for the missing userContext when the event object is created.

Reply:
Works for me.  Done.

Reply:
> `authenticatedUser.getAuthenticationProvider().getUserContext(authenticatedUser)`

I'm worried about that approach.

The `UserContext` is effectively the data side of a user's session, and typically is only created once, with that instance existing until the user logs out. If an extension implementation needed to persist some sort of state throughout the user's session, the lifecycle of that state would be tied to the `UserContext`. There's also the new `invalidate()` function to consider, which will need to be properly invoked for that short-lived `UserContext`.

This could have performance consequences down the line.

Reply:
Is there an expectation that this getUserContext() method in the event handler will get called repeatedly?

I can revert back to the original method of storing it there - but, if we do that, we have to handle the exception that happens when it is null.  I'm happy to put back my code for just logging the exception and not passing it through in this instance, if that's okay, or I'm happy to investigate other routes if those are preferred?

Reply:
Since `getUserContext` looks like an accessor method, it was an (obviously incorrect) assumption that a provider would create it once and return the same reference for a given authenticated user thereafter

Reply:
Okay, reverted to having the objects in here, but took a slightly different angle on it by having the `AuthenticationSuccessEvent` constructor handle pulling all of the objects out of the `AuthenticatedUser` object, and handled catching the `GuacamoleResourceNotFoundException` exception there, instead of in `AuthenticationService`.  Let me know if this looks okay, or if it should go back to the way I originally had it, catching the error in `AuthenticationService`, instead.  Or if there's some other way that it needs to be done?